src/simple_UART.c

```diff
@@ -5,21 +5,29 @@
  *      Author: abates
  */
 
+
 #include "simple_UART.h"
-//#include "buffer.h"
-//#include "FSM.h"
-//#include "LEDs.h"
+#include "stm32f4xx_dma.h"
 #include "FreeRTOS.h"
 #include "Task.h"
+#include "Buffer.h"
 #include <string.h>
 
-char stringtosend[MAX_OUPUT_DATA] = "";
-char UARTInput[MAX_BUFFER_DATA];
-uint8_t UARTInputIndex = 0;
+//Register bit for enabling TXEIE bit. This is used instead of the definitions in stm32f4xx_usart.h
+#define USART_TXEIE	0b10000000
+#define USART_RXEIE	0b100000
```

Comment:
Why aren't you using the definitions in stm32f4xx_usart.h?

Reply:
I didn't use the definitions in stm32f4xx_usart.h because it wasn't working. It wasn't toggling the correct bit.

Reply:
Cool cool, works for me!